src/engine/client/serverbrowser.cpp

```diff
@@ -484,7 +484,9 @@ void CServerBrowser::SetInfo(int ServerlistType, CServerEntry *pEntry, const CSe
 	pEntry->m_Info.m_Flags &= FLAG_PASSWORD;
 	if(str_comp(pEntry->m_Info.m_aGameType, "DM") == 0 || str_comp(pEntry->m_Info.m_aGameType, "TDM") == 0 || str_comp(pEntry->m_Info.m_aGameType, "CTF") == 0 ||
 		str_comp(pEntry->m_Info.m_aGameType, "SUR") == 0 ||	str_comp(pEntry->m_Info.m_aGameType, "LMS") == 0)
-		pEntry->m_Info.m_Flags |= FLAG_PURE;
+		if(pEntry->m_Info.m_MaxPlayers <= MAX_PLAYERS)
+			pEntry->m_Info.m_Flags |= FLAG_PURE;
```

Comment:
I think servers aren't allowed to display as DM/TDM/CTF/... if they contain modified code. Clearly, a server with >16 players would have to be a mod under disguise. That's not cool, those are banned from the master server, so maybe just drop that server (and notify it as a `dbg_msg`?)

Reply:
It's the job of the masterserver to not make this server in the server list.
I think it's fine to show them as long as there are not flagged as pure.
The discussion is open.

Reply:
Fair enough. @oy thoughts?

Reply:
As it breaks current compatibility, I'd like to pospone this to 0.8. The whole thing has to be pretty tight.

Reply:
Not that I want this feature, but there might be a bunch of features that we want to fix/add/mature in 0.7.1. Breaking compatibility might come from others, like gametype naming so I think it might not be a issue to worry about atm.
But indeed the whole thing has to be made carefully.

Reply:
Well, yeah depends on what gets added.
But in general it should be like it used to be. We stick to a maximum of 16 players, so it should just be able to handle 16 and drop everything that validates that like before.

Reply:
@oy: what about the design choice we were discussing in that reviewing thread? Keep modified DM/TDM/CTF servers & display as unpure or drop them?

Reply:
@oy you mean dropping everything with max_player > 16 in engine/client/client.cpp
https://github.com/teeworlds/teeworlds/blob/9eeb7dffa5826c38c7b5ef70cb787a3dd65bc3e1/src/engine/client/client.cpp#L869-L872

Reply:
and/or do it totally differently without increasing the capacity of the client to render players ? (sticking to 16 max)

Reply:
@Dune-jr 
Drop them.
@nheir 
Yeah drop that. Keep it consistent.